Core/Aspects/Validation/ValidationAspect.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Castle.DynamicProxy;
+using FluentValidation;
+using WebNetSample.Core.CrossCuttingConcerns.Validation;
+using WebNetSample.Core.Utilities.Interceptors;
+
+namespace WebNetSample.Core.Aspects.Validation;
+
+// Validation implementation as AOP design
+public class ValidationAspect : MethodInterception
+{
+    private readonly Type _validatorType;
+
+    public ValidationAspect(Type validatorType)
+    {
+        if (!typeof(IValidator).IsAssignableFrom(validatorType))
+            throw new Exception("This is not a validation class.");
```

Comment:
Probably this is not your full code, but I advise to **always** add curly brackets for conditions/loops, it will simplify reading